From 264f1b31ca5af49ba1ce7a7346c4af7f1fe62867 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?David=20Nystr=C3=B6m?= <david.nystrom@est.tech>
Date: Mon, 10 Nov 2025 12:08:54 +0100
Subject: [PATCH] Fix dependencies to use system
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: David Nyström <david.nystrom@est.tech>
Upstream-Status: Inappropriate [oe specific]
---
 thirdparty/dependencies.cmake | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/thirdparty/dependencies.cmake b/thirdparty/dependencies.cmake
index 3d7f137ec9..7037da5d26 100644
--- a/thirdparty/dependencies.cmake
+++ b/thirdparty/dependencies.cmake
@@ -275,9 +275,13 @@ endif()
 # Gflags
 #

-if(ENABLE_SAMPLES OR ENABLE_TESTS)
-    add_subdirectory(thirdparty/gflags EXCLUDE_FROM_ALL)
-    ov_developer_package_export_targets(TARGET gflags)
+if(ENABLE_SYSTEM_GFLAGS)
+    find_package(gflags REQUIRED)
+else()
+    if(ENABLE_SAMPLES OR ENABLE_TESTS)
+        add_subdirectory(thirdparty/gflags EXCLUDE_FROM_ALL)
+        ov_developer_package_export_targets(TARGET gflags)
+    endif()
 endif()

 #
